swan-lake/development-tutorials/source-code-dependencies/customize-formatting.md

```diff
@@ -0,0 +1,409 @@
+---
+title: Customize formatting
+description: The section on providing custom configurations to the formatter.
+keywords: ballerina, programming language, formatting, custom configurations
+permalink: /learn/customize-formatting/
+active: customize-formatting
+intro: This feature allows developers to customize how a Ballerina project is formatted via a local or remote configuration file. This allows for consistency in code style across projects in an organization and simplifies the process of enforcing formatting standards. This is introduced as an experimental feature in Ballerina 2201.9.0.
+---
+
+## The formatting configuration file
+
+Users can customize formatting configuration via a configuration file in the `toml` format. Users can provide an arbitrary name for the file.
+
+The formatting configuration file can be specified in two ways:
+
+* `Ballerina.toml`: Define the path to the configuration file in the `Ballerina.toml` file. The path can be absolute, relative to the package root, or a URL (HTTP/HTTPS). This path has to be specified under the `format` header in the `Ballerina.toml` file as `configPath`.
+
+    ```toml
+        [package]
+        org = "format-org"
+        name = "format_project"
+        version = "0.1.0"
+        distribution = "2201.9.0"
+
+        [format]
+        configPath = "Format.toml"
+    ```
+
+* `Format.toml`: If no configuration path is specified in the `Ballerina.toml` file, the formatter will look for a file named `Format.toml` in the package's root directory.
+
+If the `configPath` value is not provided in the `Ballerina.toml` file or the `Format.toml` file does not exist in the project root, default formatting options will be used.
+
+If the format `configPath` value is an HTTP/HTTPS URL, the configuration file will be cached in the `target` directory. If the remote configuration file is modified, then make sure to delete the cached `Format.toml` file inside the `target/format/` directory.
+
+## Formatting Options
+
+### Indent
+
+#### indentSize
+
+* Specifies the number of spaces for each level of indentation.
+* Value: Integer (e.g., 2, 4)
+* Default: 4
+
+    ```ballerina
+    public function main() {
+        boolean condition = true;
+        if condition {
+            io:println("Indented code");
+        }
+    }
+    ```
+
+#### continuationIndentSize
+
+* Determines the indentation size for continuation lines.
+* Value: Integer (e.g., 4, 8)
+* Default: 8
+
+    ```ballerina
+    function longFunctionName(int param1, int param2, int param3,
+            int param4, int param5) {
+    }
+    ```
+
+### Wrapping
+
+#### maxLineLength
+
+* Maximum line length before code is wrapped.
```

Comment:
There are some inconsistencies between the descriptions of the properties. In the `continuationIndentSize`, you describe what the property does, while in this case, you simply describe the property. It's better to stick to one writing style, preferably the former.